src/System.Resources.Extensions/src/System/Resources/Extensions/DeserializingResourceReader.cs

```diff
@@ -38,12 +39,72 @@ private object ReadBinaryFormattedObject()
         {
             if (_formatter == null)
             {
-                _formatter = new BinaryFormatter();
+                _formatter = new BinaryFormatter()
+                {
+                    Binder = new UndoTruncatedTypeNameSerializationBinder()
+                };
             }
 
             return _formatter.Deserialize(_store.BaseStream);
         }
 
+
+        internal class UndoTruncatedTypeNameSerializationBinder : SerializationBinder
+        {
+            public override Type BindToType(string assemblyName, string typeName)
+            {
+                Type type = null;
+
+                // determine if we have a mangled generic type name
+                if (!AreBracketsBalanced(typeName))
+                {
+                    // undo the mangling that may have happened with Desktop's bad
+                    // ResXSerialization binder.
+                    typeName = typeName + ", " + assemblyName;
+
+                    type = Type.GetType(typeName);
```

Comment:
> type = Type.GetType(typeName); [](start = 19, length = 31)

I assume it is ok if this throw if for any reason we got malformed type name. right?

Reply:
Ok looks like BinaryFormatter was a bit different.  I’ll use it’s TypeLoading algo as precedent rather than Resx since resx was never happening at runtime.

Reply:
Acutally, I take that back.  It sepecifies `throwOnError:false, ignoreCase:false` but those are the defaults.  https://referencesource.microsoft.com/#mscorlib/system/type.cs,105, https://github.com/dotnet/coreclr/blob/cea2bba3c295c5280d01900f7ae5492951ad8995/src/System.Private.CoreLib/src/System/Type.CoreCLR.cs#L41

This will be equivalent, and will most closely match what ResXSerializationBinder was doing.  If we don't throw here, the same data will be used by BinaryFormatter and will throw as well.

Reply:
I am fine with that. I was trying to be explicit about the expectation when running into this case.